packages/synthetics-sdk-broken-links/src/broken_links.ts

```diff
@@ -138,6 +140,9 @@ export async function retrieveLinksFromPage(
   get_attributes: string[]
 ): Promise<LinkIntermediate[]> {
   const origin_url = await page.url();
+  // `page.evaluate()` runs in a different context, isolated from main Node.js,
```

Comment:
If this is the case, I think that we should consider moving the callback function provided to page.evaluate to its own function to be tested. There's a decent amount of logic here that we should ensure is tested.

Reply:
I can do that but just to clarify the logic is tested in broken_links.spec.ts but its not "seen" by c8 since its in a different context. This means that we are not 100% sure that all lines are covered but testing for the function exists.

Reply:
Yes, so I understand that completely - what you're running the risk of is that when this function is changed even further, there's no way to trust / know that we're actually covering new logic.

Reply:
Not in this CL necessarily, but I think that what you should do is to pull this out, test it at a functional level, and then pass it in with a more integration style test that uses it in the context of puppeteer.

Reply:
agreed. I will add it to the project hotlist!